scripts/image-tester/sphinx_alt_text_validator/verify_images.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Utility script to verify that all images have alt text"""
+
+from pathlib import Path
+
+
+def is_image(line: str) -> bool:
+    """Determine if a line is an image"""
+    return line.strip().startswith((".. image:", ".. plot:"))
+
+
+def is_option(line: str) -> bool:
+    """Determine if a line is an option"""
+    return line.strip().startswith(":")
+
+
+def is_valid_image(options: list[str]) -> bool:
+    """Validate one single image"""
+    alt_exists = any(option.strip().startswith(":alt:") for option in options)
+    nofigs_exists = any(option.strip().startswith(":nofigs:") for option in options)
+
+    # Only `.. plot::`` directives without the `:nofigs:` option are required to have alt text.
+    # Meanwhile, all `.. image::` directives need alt text and they don't have a `:nofigs:` option.
+    return alt_exists or nofigs_exists
+
+
+def validate_image(file_path: str) -> tuple[str, list[str]]:
```

Comment:
Same function but removing this conditional:
```python
if file_path in ALLOWLIST_MISSING_ALT_TEXT:
    return [file_path, []]
```

Now the function gets the files filtered from `__init__.py` so no need to verify any allowlist anymore